content/matrices/components/gauss-solver.ts

```diff
@@ -0,0 +1,196 @@
+        }
+      });
+    }
+
+    const $arrows = this.$$('.connections-right polygon');
+    const outRows = ['outRow1', 'outRow2'] as ('outRow1'|'outRow2')[];
+
+    for (const [i, key] of outRows.entries()) {
+      slide($arrows[i], {
+        move: (p: Point) => {
+          // Check that this row not equal to the other output
+          const row = clamp(Math.round((p.y - 20) / 40), 0, this.size - 1);
+          this.model[key] = row;
+        }
+      });
+    }
+
+    // Watch for change and update display of output cells
+    this.model.watch((state) => {
+
+      console.log(`State: {
+          op: ${state.op},
+          factor: ${state.factor}, factorString: ${state.factorString},
+          inRow1: ${state.inRow1}, inRow2: ${state.inRow2},
+          outRow1: ${state.outRow1}, outRow2: ${state.outRow2}
+        }`);
+      for (const [i, row] of this.$outputCells.entries()) {
+        for (const [j, val] of row.entries()) {
+
+          switch (state.op) {
+            case 'multiply':
+              // "Multiply a row by a non-zero factor"
+              if (state.factor && i === state.outRow1) {
+                const m = Expression.parse(this.$inputCells[state.inRow1][j].text).evaluate();
+                val.text = '' + (state.factor * m);
+              } else {
+                val.text = this.$inputCells[i][j].text;
+              }
+              break;
+            case 'add':
+              // "Add a multiple of one row to another"
+              val.text = this.$inputCells[i][j].text;
+              if (state.factor && i === state.outRow1) {
+                const c1 = Expression.parse(this.$inputCells[state.inRow1][j].text).evaluate();
+                const c2 = Expression.parse(this.$inputCells[state.inRow2][j].text).evaluate();
+                val.text = '' + (c1 + state.factor * c2);
+              } else {
+                val.text = this.$inputCells[i][j].text;
+              }
+              break;
+            case 'swap':
+              // "Swap two rows"
+              if (i === state.outRow1) {
+                val.text = this.$inputCells[state.inRow1][j].text;
+              } else if (i === state.outRow2) {
+                val.text = this.$inputCells[state.inRow2][j].text;
+              } else {
+                val.text = this.$inputCells[i][j].text;
+              }
+              break;
+          }
+        }
+      }
+    });
+
+    // Button to Apply Row Operation
+    const $applyBtn = this.$('.apply');
+    $applyBtn?.on('click', () => {
+      // swapping input and output
+      for (const [i, row] of this.$inputCells.entries()) {
+        for (const [j, val] of row.entries()) {
+          val.text = this.$outputCells[i][j].text;
+        }
+      }
+      this.inputStack.push(this.$outputCells.map(row => row.map(val => val.text)));
+      console.log(this.inputStack);
+
+      if (this.checkForSolvedIdentity()) {
+        console.log('Success!');
+        this.$step.tools.confetti();
```

Comment:
Rather than showing confetti, maybe something more subtle like `this.$step.addHint('correct')`

Reply:
That's definitely more appropriate, I just really wanted a chance to use the confetti 😆